### PR TITLE
node event docs

### PR DESCRIPTION
**Pull request by anhou**
along with PR  https://github.com/RackHD/on-core/pull/178 https://github.com/RackHD/on-tasks/pull/279 https://github.com/RackHD/on-http/pull/368 , other PRs had been merged 'discovered 'https://github.com/RackHD/on-tasks/pull/240  and 'removed' https://github.com/RackHD/on-http/pull/339

@RackHD/corecommitters

### CODE REVIEW

**Comment by amymullins** [comment]
Hi @anhou , this documentation references nodeTypes of pdu, mgmt, and enclosure in addition to the roadmap features of compute (https://github.com/RackHD/RackHD/wiki/proposal-server-node-event) and switch  (https://github.com/RackHD/RackHD/wiki/proposal-switch-node-event).  Will events for pdu, mgmt. and enclosure be developed/tested as part of the referenced PR ( https://github.com/RackHD/on-core/pull/177) and future PRs for the remaining event types ?

**Comment by anhou** [comment]
@amymullins  Yes, the ORFS246/247 is only for compute/switch node, but in the implementation, I extend it to all available node types in RackHD, to achieve the functionality for all types of nodes with a common way, to adapt possiable future requirements for pdu/mgmt/enclosure nodes.  
 
The remaining events has been achieved by previous PRs for 'discovered' https://github.com/RackHD/on-tasks/pull/240  for 'removed', https://github.com/RackHD/on-http/pull/339  as a part of https://github.com/RackHD/RackHD/wiki/proposal-AMQP-notifications-on-node-discovery-or-inaccessibility, it's a little duplicated with ORFS 246/247. but we considered them together

**Comment by amymullins** [comment]
@anhou just wanted to make sure that if we are documenting the nodeTypes not covered by the ORFSs that they are actually supported.  You confirmed a common development approach was used, but will they be tested/validated as part of this effort?

**Comment by anhou** [comment]
@amymullins  missed tested/validated part :)  I think the test cases are designed by ORFS, not by document, so IMHO, the tests should be based on switch/server ORFS. the additional nodeType support  wouldn't influence original test plan.

**Comment by amymullins** [comment]
Maybe remove those nodeTypes from the "Supported" Node types until they can be validated/tested?  Otherwise, seems kind of like false advertising as it is documented to be supported but never tested/validated:)

**Comment by anhou** [comment]
@amymullins  make sense, I just update the docs based on the discussion result, replace original state with 'action' also it's similar with github event API https://developer.github.com/v3/activity/events/types/#pullrequestevent and google map event API . I also remove the nodeType in column, because all the contents are 'action' related. please review,

File: docs/rackhd/notification.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+|                 |           |                                  | relationships with other       |
+|                 |           |                                  | models, etc.                   |
++-----------------+-----------+----------------------------------+--------------------------------+
+| sku.assigned    | one-shot  | Event occurs when node's `sku`   | Triggered when sku is created  |
+|                 |           | field is assigned.               | and associated with the node   |
++-----------------+-----------+----------------------------------+--------------------------------+
+| sku.unassigned  | one-shot  | Event occurs when node's `sku`   | Triggered when sku is          |
+|                 |           | field is unassigned.             | not associated with the node   |
++-----------------+-----------+----------------------------------+--------------------------------+
+| sku.updateed    | one-shot  | Event occurs when node's `sku`   | Triggered when node's sku      |
```
Comment (yyscamper) [inline docs/rackhd/notification.rst]: typo: updated

File: docs/rackhd/notification.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+|                 |           | field is updated.                | s associated with another sku  |
++-----------------+-----------+----------------------------------+--------------------------------+
+| obms.assigned   | one-shot  | Event occurs when node's `obms`  | Triggered when obm is created  |
+|                 |           | field is assigned.               | and associated with the node   |
++-----------------+-----------+----------------------------------+--------------------------------+
+| obms.unassigned | one-shot  | Event occurs when node's `obms`  | Triggered when obm is          |
+|                 |           | field is unassigned.             | not associated with the node   |
++-----------------+-----------+----------------------------------+--------------------------------+
+| obms.updateed   | one-shot  | Event occurs when node's `obms`  | Triggered when node don't have |
```
Comment (yyscamper) [inline docs/rackhd/notification.rst]: typo: updated

Comment (yyscamper) [inline docs/rackhd/notification.rst]: It's better to change the "when-trigger" description:
```
Triggered when all obm settings are removed from a node.
```

Comment (yyscamper) [inline docs/rackhd/notification.rst]: Another question, if this event is  triggered when all obm settings are removed, isn't more reasonable to change the `action` to "obm.removed"

File: docs/rackhd/notification.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+========= ====== =================================
+type      String The event type, it could be `node`, `sku`, etc.
+action    String The `action` that was performed, it's related to event `type`
+========= ====== =================================
+
+
+Node Events
+~~~~~~~~~~~~~~~~~~~~
+
+An example of an minimal node event message format is below:
```
Comment (yyscamper) [inline docs/rackhd/notification.rst]: typo "an minimal" ==> "a minimal"

**Comment by WangWinson** [comment]
:+1:

**Comment by pengz1** [comment]
+1

**Comment by yyscamper** [comment]
:+1:

**Pull request merged**